### DISCUSSION TITLE
How to calculate the mass of the battery?

### DISCUSSION THREAD

**Discussion by tGisbertz** [pybamm-team/PyBaMM#2647] [Q&A] [answered]
Hello everyone,

I want to calculate the energy density of the battery for different parameter variations (parameter set OKane2022). For this I obviously need to calculate the battery mass, for which I wanted to do something like that (index f is for bindes):
<img width="282" alt="image" src="https://user-images.githubusercontent.com/85396095/216106541-f9dfde0b-e95d-498c-8820-1a3c22ee151a.png">
However there is no electrolyte density (or is there in the parameter set). It seems like a dumb question, but what is the best way to calculate the battery mass in pybamm?

Thanks in advance!

**Comment by valentinsulzer** [comment, marked as answer]
Interesting question, we don't have any of this kind of analysis in PyBaMM but maybe we should. In the meantime, there are a couple of libraries that you could draw inspiration from for doing it yourself https://github.com/mjlacey/cellmodels, https://github.com/ndrewwang/BotB

Generally I think your approach is correct, you just need to look up values for the missing parameters

**Reply by brosaplanella** [reply]
If it helps, we collected some parameter values in our article [O'Regan et al (2022)](https://www.sciencedirect.com/science/article/pii/S0013468622008593). It is not open access but the [preprint here](https://chemrxiv.org/engage/api-gateway/chemrxiv/assets/orp/resource/item/61e2fd304a603d7d99314784/original/thermal-electrochemical-parameters-of-a-high-energy-lithium-ion-cylindrical-battery.pdf) contains the same info.
